src/box/lua/load_cfg.lua

```diff
@@ -962,13 +964,40 @@ local function load_cfg(cfg)
             box[k] = v
         end
     end
+
     setmetatable(box, nil)
-    box.cfg = setmetatable(cfg,
+    -- There is only a single way for setting a value to a box option -
+    -- call 'box.cfg{ option_name=option_value }'. Direct modifications
+    -- of table box.cfg are not allowed. On attempt to make a direct
+    -- modifications (setting a new option or updating an existed one) we raise
+    -- an error. For implementation this we are using metamethods __newindex
+    -- and __index. Both __index and __newindex are relevant only when the
+    -- index does not exist in the table. The only way to catch all accesses to
+    -- a table is to keep it empty. So we use an empty proxy table and proxying
+    -- setting and reading to another table.
+    raw_cfg = table.copy(cfg)
+    local proxy_table = {}
+    box.cfg = setmetatable(proxy_table,
         {
-            __newindex = function(table, index) -- luacheck: no unused args
+            -- tarantoolctl (extra/dist/tarantoolctl.in) expects a metatable
+            -- with metamethods, not a string. Thus we couldn't use __metatable
+            -- instead.
+            __name = 'box_cfg',
+            __index = raw_cfg,
+            __newindex = function(self, key) -- luacheck: no unused args
+                if template_cfg[key] ~= nil then
+                    error('Use function box.cfg for modifications')
+                end
                 error('Attempt to modify a read-only table')
```

Comment:
Why do we need to differ these two cases? I believe `box.cfg` modification is forbidden, whether the key exists in the table or not.

Reply:
Yeah, nice change, thanks.